src/attack.rs

```diff
@@ -152,15 +164,17 @@ fn attack_damage_system(
 
 fn breakable_system(
     mut events: EventReader<CollisionEvent>,
-    mut despawn_query: Query<&mut Breakable>,
+    mut despawn_query: Query<(&mut Breakable, Option<&mut Drop>)>,
     mut commands: Commands,
 ) {
     for ev in events.iter() {
         if let CollisionEvent::Started(e1, e2, _flags) = ev {
             for e in [e1, e2].iter() {
-                if let Ok(mut breakable) = despawn_query.get_mut(**e) {
+                if let Ok((mut breakable, drop)) = despawn_query.get_mut(**e) {
                     if breakable.hit_count < breakable.hit_tolerance {
                         breakable.hit_count += 1;
+                    } else if let Some(mut drop) = drop {
+                        drop.drop = true;
```

Comment:
As I said I would think of throwing an event here but not sure how the others feel about that.

Reply:
An event makes sense to me here.

We use a similar pattern with the attack collision system, so at least there's some precedent, too. The system responsible for detecting a certain kind of collisions emits an event when one occurs so other systems can supplement the behavior.